### PR TITLE
[DO NOT MERGE] - Redux undo 

### PR DESCRIPTION
**Pull request by gavinhenderson**
Hey @shayc and @martinbedouret was wondering if you would like to take a look at my progress and give some feedback. Currently there is no undoing but the 'undo' button now appears on the notifications that it should. 

I added 'action' to the Notification component as that is what 'Snackbar' takes.

Thoughts?

### CODE REVIEW

File: src/components/Board/Board.container.js [hunk 1]
```diff
@@ -360,8 +361,19 @@ export class BoardContainer extends Component {
   };
 
   handleEditTileEditorSubmit = tiles => {
-    const { board, editTiles } = this.props;
+    const { intl, board, editTiles, showNotification } = this.props;
+
     editTiles(tiles, board.id);
+    showNotification(intl.formatMessage(messages.tilesEdited), [
+      <Button
```
Comment (shayc) [inline src/components/Board/Board.container.js:368]: I would refactor this `Button` component to its own `UndoButton` in our shared UI components folder. This seems generic enough and many components will want to use this and it's nice to isolate the translation there as well.

File: src/components/Board/Board.container.js [hunk 2]
```diff
@@ -449,7 +461,16 @@ export class BoardContainer extends Component {
     const { intl, deleteTiles, showNotification, board } = this.props;
     deleteTiles(this.state.selectedTileIds, board.id);
     this.setState({ selectedTileIds: [] });
-    showNotification(intl.formatMessage(messages.tilesDeleted));
+    showNotification(intl.formatMessage(messages.tilesDeleted), [
+      <Button
```
Comment (shayc) [inline src/components/Board/Board.container.js:465]: Into a single component `UndoButton`

File: src/components/Notifications/Notifications.container.js [hunk 3]
```diff
@@ -19,9 +24,13 @@ class NotificationsContainer extends Component {
...
     /**
-     * Hids notification bar
+     * Hides notification bar
+     */
+    hideNotification: PropTypes.func.isRequired,
+    /**
+     * The actions at the end of the notification
      */
-    hideNotification: PropTypes.func.isRequired
+    action: PropTypes.arrayOf(PropTypes.object)
```
Comment (shayc) [inline src/components/Notifications/Notifications.container.js:33]: type should be: `PropTypes.node`

```
  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.
  optionalNode: PropTypes.node,
```

File: src/components/Notifications/Notifications.component.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ const propTypes = {
   config: PropTypes.object.isRequired,
   handleNotificationDismissal: PropTypes.func.isRequired,
   message: PropTypes.string.isRequired,
+  action: PropTypes.arrayOf(PropTypes.object),
```
Comment (shayc) [inline src/components/Notifications/Notifications.component.js:10]: Type should be: `PropTypes.node`

```
// Anything that can be rendered: numbers, strings, elements or an array
// (or fragment) containing these types.
optionalNode: PropTypes.node,
```

File: src/components/Board/Board.messages.js [hunk 5]
```diff
@@ -1,6 +1,14 @@
 import { defineMessages } from 'react-intl';
 
 export default defineMessages({
+  undo: {
+    id: 'cboard.components.Board.undo',
```
Comment (shayc) [inline src/components/Board/Board.messages.js:5]: When refactoring to `UndoButton`, should update this string:
` id: 'cboard.components.UndoButton.undo'`

**Review by shayc — CHANGES_REQUESTED** [review]
I like your direction, great start!

**Comment by gavinhenderson** [comment]
@shayc Thanks for the feedback! I will make those changes and get the functionality working.

Ill close this PR just now and open a new one when im done

**Pull request closed without merging**